pkg/observer/metrics.go

```diff
@@ -53,6 +57,32 @@ func newMetrics() *metrics {
 			},
 			[]string{"observer"},
 		),
+
+        // Visibility metrics for priority flow
+        ChangedNodeGroups: prometheus.NewCounterVec(
+            prometheus.CounterOpts{
+                Name:      "changed_nodegroups",
+                Namespace: metricsNamespace,
+                Help:      "counter of nodegroups detected as changed per run",
+            },
+            []string{},
+        ),
+        ApplyingNodeGroups: prometheus.NewCounterVec(
+            prometheus.CounterOpts{
+                Name:      "applying_nodegroups",
+                Namespace: metricsNamespace,
+                Help:      "counter of nodegroups applied per run (lowest priority batch)",
+            },
+            []string{},
+        ),
+        BlockedNodeGroups: prometheus.NewCounterVec(
```

Comment:
note: I'm not entirely sure how well these metrics will work but I guess we'll see once we launch the change. Better to have them than not, but I'm sure we can make tweaks to their usefulness later.

Reply:
@dnorman3 could you run this in your lab and write a page showing what the metrics look like so we can see before merging this change?